### ISSUE TITLE
[question] `conan install --update` doesn't seem to get the latest revision.

### ISSUE THREAD

**Issue by birsoyo** [conan-io/conan#12835]
Is there some additional command line or setting so that when `conan install --update` is issued, the latest revision is downloaded to local cache?

This is for 1.55. 

Also, just in case it is relevant. The revision mode is set to scm (i.e. ` revision_mode = "scm"`)

For example: 
To check the local cache revision I do:
`conan search <package> -rev` 
returns
```
Revisions for '<package>':
769be40fc79547aa1b091c78e93119eec0989b86 (2022-12-28 08:23:46 UTC)
```

Now for the server:
`conan search --remote <remote> <package> -rev`
I get:
```
Revisions for '<package>' at remote '<remote>':
0f9c76286017bfb9b6aa501f9f2bcb200a642d65 (2022-12-30 08:23:49 UTC)
7e48aef42961db4bee17df89497d4d1fd21d1609 (2022-12-29 08:23:49 UTC)
769be40fc79547aa1b091c78e93119eec0989b86 (2022-12-28 08:23:46 UTC)
c8b21d643f787f7daa9122e6e5bd1bbe7dec6327 (2022-12-27 08:24:07 UTC)
32c51fe76be4ac5042214ebd4e6a4f9b9778b3bf (2022-12-21 08:24:10 UTC)
fd5c68be4b01e5665c9e027686bb63f5feec76b7 (2022-12-20 08:24:05 UTC)
548bc9859a932706432976d892aa42a58c6bff94 (2022-12-19 08:23:35 UTC)
aa1062dcb63b8f54eba930804f203a9faaaadbbf (2022-12-16 08:23:53 UTC)
0 (2022-12-15 17:09:36 UTC)
12cd3785c90cd3c01f9683a5ec89b5b6cf7346cd (2022-12-15 02:51:23 UTC)
bef7c45a252279b9aa1929bfe1e006eb8ad9efa8 (2022-12-12 08:24:06 UTC)
```

with `conan install --update` I was expecting the latest revision (i.e. `0f9c762...`) to be downloaded to local cache.

**Comment by czoido** [comment]
Hi @birsoyo, looks like this is not directly related with cmake-conan, transferring to the main conan repo

**Comment by czoido** [comment]
Hi @birsoyo,

Thanks for the question, I think that should work. For example, starting from an empty cache, let's install one old revision, then use `--update` to update to the latest.

```
➜ conan search zlib/1.2.11@ -rev -r conancenter 
Revisions for 'zlib/1.2.11' at remote 'conancenter':
ffa77daf83a57094149707928bdce823 (2022-11-02 13:46:53 UTC)
8b23adc7acd6f1d6e220338a78e3a19e (2022-10-19 09:19:10 UTC)
ce3665ce19f82598aa0f7ac0b71ee966 (2022-10-14 11:42:21 UTC)
31ee767cb2828e539c42913a471e821a (2022-10-12 05:49:39 UTC)
d77ee68739fcbe5bf37b8a4690eea6ea (2022-08-05 17:17:30 UTC)
6579cd83d7728bdb0df70b512d9b1a11 (2022-05-13 13:55:39 UTC)
95482d4fb614fadd27775f2def0a5323 (2022-05-09 06:59:58 UTC)
6fcaa61190cb5889704d322e95698630 (2022-03-29 15:47:45 UTC)
4b38406da00104befece594b529fd155 (2022-03-17 13:48:56 UTC)
b46bf663179824c23085bb71666ba8e6 (2022-01-25 01:25:38 UTC)
683857dbd5377d65f26795d4023858f9 (2021-12-14 16:35:09 UTC)
...


➜  conan install zlib/1.2.11@#95482d4fb614fadd27775f2def0a5323  --build=missing

...                                     
zlib/1.2.11: Downloaded recipe revision 95482d4fb614fadd27775f2def0a5323
...

➜ conan search zlib/1.2.11@ -rev
Revisions for 'zlib/1.2.11':
95482d4fb614fadd27775f2def0a5323 (2022-05-09 06:59:58 UTC)

➜ conan install zlib/1.2.11@ --update --build=missing
...
zlib/1.2.11: Downloaded recipe revision ffa77daf83a57094149707928bdce823
...

➜ conan search zlib/1.2.11@ -rev                     
Revisions for 'zlib/1.2.11':
ffa77daf83a57094149707928bdce823 (2022-11-02 13:46:53 UTC)
```

Maybe you have some typo? Can you provide more specific details so we can reproduce?

Thanks a lot

**Comment by birsoyo** [comment]
I followed your steps. It works for zlib in conan center. 
But it doesn't work for the package I am testing with in our artifactory server. 

I am investigating what might be different.
Any suggestions?

**Comment by czoido** [comment]
Hi @birsoyo,
Could it be that you have revisions disabled in the client you are using? What is the output of this command?

```
conan config get general.revisions_enabled
```

Could you provide the exact set of commands with the arguments that you are running?
Also, you could try to create a new package using something like `conan new mylib/1.0 --template cmake_lib` and testing with that.

Thanks!

**Comment by birsoyo** [comment]
I get `1` back.
```
PS C:\Users\OBirsoy> conan config get general.revisions_enabled
1
```

**Comment by birsoyo** [comment]
These are the exact commands.
`conan install skore/develop@sketchup/rolling#c8b21d643f787f7daa9122e6e5bd1bbe7dec6327 -s compiler.version=14`
`conan install skore/develop@sketchup/rolling --update -s compiler.version=14`

After update:
 ```
 PS C:\Users\OBirsoy> conan search skore/develop@sketchup/rolling -rev
Revisions for 'skore/develop@sketchup/rolling':
c8b21d643f787f7daa9122e6e5bd1bbe7dec6327 (2022-12-27 08:24:07 UTC)
```
The revisions on the server is in my first post. "c8b21d" is not the latest.

**Comment by birsoyo** [comment]
I created a new package as you described. 
TL;DR; it works as expected.

1 change I made was to add `revision_mode="scm"`.
I created a conan package with `conan create ....` and upload it to our artifactory.
Made several changes to cpp file.

Created 3 revisions that matches the git commit hashes. 

Installing specific older revisions and later installing with an `--update` works as expected.

Note: That it is not just my machine.
It happens on my colleagues' machines on windows and macOS as well.

**Comment by czoido** [comment]
Hi @birsoyo thanks a lot for all the tests and providing the data. A few more questions:

- Do you have multiple remotes? If so, does it still happen if specify the -r <remote> in the conan install command?
- What Artifactory version are you using? 
- What kind of repository are you using? Is it local? virtual?
- Could you please provide the first part of the output of the Conan install command that does not succeed on installing the latest revision?

**Comment by birsoyo** [comment]
After some debugging yesterday, I feel like my problem might be a little bit simpler. 
It looks like, in order to update, conan does extra checks (like checking parts of the content in export/conanmanifest.txt), even though the index.json (under skore/develop/rolling) has a different revision hash (git scm hash) and its date is different. 

Having a different conan revision hash with a later date is not enough for conan to update to that later revision. There has to be a hash change in the export/conanmanifest.txt (and also, I think the hashes in export_sources folder contents as well).

In our case, the hashes might not change, for a while (days, weeks), because most of the changes are elsewhere (i.e. packaged binaries etc). skore/develop@sketchup/rolling currently updates every day, but the end goal for it to be updated quite frequently (say 10, 20 times a day). We want the new update to be picked up by everybody (including some CI builds), without additional changes elsewhere (i.e. their source code), just `conan install --update`.  

Simplest solution I can think of is to add a file to "export_sources" that changes on every commit no matter what. For example, a generated text file that contains the current git hash (or something similar). 
I think this will force conan to update to latest because, there will be an artificial change in the manifest files (in export or export_sources folders). 

Note that we didn't have this problem pre-revisions (i.e. revision of "0").

**Comment by memsharded** [comment]
Sorry this was not followed up back then.

Conan 2 now always uses revisions, and the behavior while resolving revisions is different to Conan 1.

Closing the ticket as outdated, but please create new tickets for any further question or issue, thanks for the feedback!
